lib/client.js

```diff
@@ -266,7 +275,11 @@ class Client extends Dispatcher {
         handler = new RedirectHandler(this, maxRedirections, opts, handler)
       }
 
+
       const request = new Request(opts, handler)
+      if (channels.dispatch.hasSubscribers) {
+        channels.dispatch.publish(opts)
```

Comment:
Maybe just publish the whole request object? That way if we need to do any worst-case-scenario last-minute monkey patching, we can do it. (Of course, we want to avoid those if possible, but who knows.)

Reply:
That request object is not what you think it is. It has almost nothing to monkey patch: https://github.com/nodejs/undici/blob/main/lib/core/request.js

Reply:
Hmm, yeah. Looking at that class, it looks pretty much like a normalizer/validator/holder for those arguments. The only other thing I can think of that we might want to do is wrap the handlers, but that would be done better elsewhere (like additional `diagnostics_channel` hooks).

Reply:
Does it mean that if APM needs to add additional header it should mutate the "opts" object?

Reply:
APMs will generally need to inject a header for distributed tracing, so we'd like a way to modify headers. Other than that, we are generally only consumers of data so mutability is usually not required.